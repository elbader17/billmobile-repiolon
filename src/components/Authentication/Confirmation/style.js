import { StyleSheet } from 'react-native';
import { widthPercentageToDP as wp, heightPercentageToDP as hp } from 'react-native-responsive-screen';
import { COLORS } from '../../../constants/colors';
import { FONTS } from '../../../constants/fonts';
import { METRICS } from '../../../constants/metrics';

const styles = StyleSheet.create({
  container:{
    height: hp('100%'),
    backgroundColor: COLORS.blue,
    paddingTop: 30,
    alignItems: 'center'
  },
  textBox: {
    backgroundColor: COLORS.white,
    fontFamily: FONTS.latoRegular,
    fontSize: FONTS.size18,
    width: wp('82%'),
    height: hp('8%'),
    paddingHorizontal: 10,
    borderWidth: 1,
    borderRadius: 5,
    borderColor: COLORS.gray,
    marginVertical: 7
  },
  textRegular14White: {
    fontFamily: FONTS.latoRegular,
    fontSize: FONTS.size14,
    color: COLORS.white,
  },
  textRegular14WhiteBold: {
    fontFamily: FONTS.latoSemiBold,
    fontSize: FONTS.size14,
    color: COLORS.white,
  },
  textRegular16White: {
    fontFamily: FONTS.latoRegular,
    fontSize: FONTS.size16,
    color: COLORS.white,
  },
  textRegular16WhiteBold: {
    fontFamily: FONTS.latoSemiBold,
    fontSize: FONTS.size16,
    color: COLORS.white,
  },
  buttonVerify: {
    backgroundColor: COLORS.red,
    height: hp('8%'),
    borderRadius: 0
  },
  textRegular14WhiteBold: {
    fontFamily: FONTS.latoSemiBold,
    fontSize: FONTS.size14,
    color: COLORS.white
  },
  headerText: {
    fontFamily: FONTS.latoRegular,
    fontSize: FONTS.size16,
    fontWeight: 'normal',
    color: COLORS.blue,
  },
  lineWhite: {
    backgroundColor: COLORS.white,
    width: wp('82%'),
    height: 1,
    marginVertical: 15
  },
});

export default styles;