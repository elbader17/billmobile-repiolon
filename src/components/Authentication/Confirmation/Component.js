import React from 'react';
import { Text, View, TextInput } from 'react-native';
import { Button } from "react-native-elements";
import style from './style';

class Confirmation extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      email: '',
      confirmPassword: '',
    };
  }

  handleConfirmationCode = () => {
    const { confirmationEmail, confirmationCode } = this.state;
    const { confirmCode } = this.props;
    confirmCode(confirmationEmail,confirmationCode,{})
  }

  setConfirmPassword = (value) => this.setState({ confirmPassword: value })

  render() {
    return(
      <View state={ style.container }>
        <View style={ style.container2 }>       
          <Text style={ style.text }>
            Verificar Cuenta
          </Text>
          <View style={ style.textBoxBtnHolder }>
            <TextInput
              label="Email"
              onChangeText={ (value) => this.setState({ confirmationEmail: value }) }
              placeholder="Tu email"
              style={ style.textBox }
            />
          </View>
          <View style={ style.textBoxBtnHolder }>
            <TextInput
              label="Codigo de confirmación"
              onChangeText={ (value) => this.setState({ confirmationCode: value }) }
              placeholder="Codigo de Confirmación"
              style={ style.textBox }
            />
          </View>
          <Button
            title='VERIFICAR'
            testID={'submitConfirmation'}
            onPress={ this.handleConfirmationCode }
            buttonStyle={ style.submit }
            titleStyle={ style.submitText }
          />
        </View>
      </View>
    )
  }
}

export default Confirmation;