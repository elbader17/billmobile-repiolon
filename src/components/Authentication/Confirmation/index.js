import { connect } from 'react-redux';
import Confirmation from './Component';
import { confirmCode } from '../../../app/authentication/actions';

const mapStateToProps = () => ({});

function mapDispatchToProps(dispatch) {
  return {
    confirmCode: (email, confirmationCode) => dispatch(confirmCode(email, confirmationCode)),
  };
}

const component = connect(
  mapStateToProps,
  mapDispatchToProps,
)(Confirmation);

export default component;
