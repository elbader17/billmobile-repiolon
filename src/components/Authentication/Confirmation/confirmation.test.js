import React from 'react';
import { shallow } from 'enzyme';
import Confirmation  from './Component';

test('render correctly component Confirmation', () => {
  //const component = shallow(<Confirmation />);
  //console.log(shallow(component.props().children({getParams: jest.fn()})));
  const navigation = { navigate: jest.fn() };

  const component = shallow(
    <Confirmation navigation={navigation} />
  );
  console.log(component.props().children());
  //console.log(shallow(component.props().children({})));
});
  