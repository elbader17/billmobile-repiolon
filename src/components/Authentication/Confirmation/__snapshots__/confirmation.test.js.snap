// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering Component Confirmation renders correctly Confirmation 1`] = `
<View
  state={
    Object {
      "height": 1334,
      "width": 750,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "paddingHorizontal": 45,
        "paddingTop": 15,
      }
    }
  >
    <Text
      style={
        Object {
          "fontFamily": "Lato-Regular",
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      Verificar Cuenta
    </Text>
    <View
      style={
        Object {
          "alignSelf": "stretch",
          "justifyContent": "center",
          "marginBottom": 6,
          "marginTop": 2,
          "position": "relative",
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        label="Email"
        onChangeText={[Function]}
        placeholder="Tu email"
        style={
          Object {
            "alignSelf": "stretch",
            "borderColor": "#C4C4C4",
            "borderRadius": 2,
            "borderWidth": 1,
            "fontFamily": "Lato-Regular",
            "fontSize": 16,
            "height": 45,
            "paddingLeft": 8,
            "paddingRight": 45,
            "paddingVertical": 0,
          }
        }
        underlineColorAndroid="transparent"
      />
    </View>
    <View
      style={
        Object {
          "alignSelf": "stretch",
          "justifyContent": "center",
          "marginBottom": 6,
          "marginTop": 2,
          "position": "relative",
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        label="Codigo de confirmación"
        onChangeText={[Function]}
        placeholder="Codigo de Confirmación"
        style={
          Object {
            "alignSelf": "stretch",
            "borderColor": "#C4C4C4",
            "borderRadius": 2,
            "borderWidth": 1,
            "fontFamily": "Lato-Regular",
            "fontSize": 16,
            "height": 45,
            "paddingLeft": 8,
            "paddingRight": 45,
            "paddingVertical": 0,
          }
        }
        underlineColorAndroid="transparent"
      />
    </View>
    <View
      style={
        Object {
          "borderRadius": 2,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="submitConfirmation"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#EE6123",
              "borderColor": "#2089dc",
              "borderRadius": 2,
              "borderWidth": 0,
              "flexDirection": "row",
              "height": 40,
              "justifyContent": "center",
              "padding": 8,
              "width": 266,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#fff",
                "fontFamily": "Lato-Regular",
                "fontSize": undefined,
                "paddingBottom": 1,
                "paddingTop": 2,
                "textAlign": "center",
              }
            }
          >
            VERIFICAR
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
